### ISSUE TITLE
The `never` option of `declaration-block-trailing-semicolon` takes away a semicolon Less requires, leaving a file Less refuses

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#309]
Less requires the semicolon that closes an `@extend`, whether or not the at-rule is the last thing in its block. `declaration-block-trailing-semicolon` under `never` takes it away:

```less
a {
	@extend .b;
}
```

comes back from `--fix` as

```less
a {
	@extend .b
}
```

and `less.render` compiles the first and refuses the second with `@extend rule is missing block or ending semi-colon`. Nothing is reported on the second run, since the rule got what it asked for.

Nothing about this turns on a comment. `a {\n\t@extend .b // c;\n}` goes the same way and is refused for the same reason, and so does an `@extend` standing behind a declaration, `a {\n\tcolor: pink;\n\t@extend .b;\n}`. The same fix under `postcss-scss` writes the same output; whether Sass minds is not settled here, since Sass lets the last statement of a block go without its semicolon in the ordinary case.

An ordinary declaration is untouched by any of this — `a {\n\tcolor: pink;\n}` becomes `a {\n\tcolor: pink\n}` and Less compiles it, which is the whole point of the option.

## Why no oracle sees it

`converge.mjs` parses the output with the syntax that wrote it, and `postcss-less` reads `@extend .b` without a semicolon perfectly well — the parser is not the compiler, and only the compiler minds. It is the shape [#231](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/231) wore under `postcss-less`, where a round trip stayed stable over a file Less could not read; there the parser disagreed with Less about a brace, here about a semicolon.

## What would end it

The option cannot be satisfied for such an at-rule, so the fix has to be declined and the warning left standing — the answer [#231](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/231), [#211](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/211) and the rest of that family give whenever an option asks for something the syntax has nowhere to put. What the rule needs is a reading of which nodes carry a semicolon the syntax will not part with; `@extend` is the one measured here, and the rest of the list is worth measuring before the guard is written rather than after.
